main.cc

```diff
@@ -537,6 +553,11 @@ void print_starting_message(int ac, char** av, const bpo::parsed_options& opts)
     fmt::print("parsed command line options: {}\n", format_parsed_options(opts.options));
 }
 
+static void checkpoint(stop_signal& stop, sstring what, bool ready = false) {
```

Comment:
nit: Can be a method of stop_signal itself, thus aligned with stop_signal::ready()

Reply:
Yeah, this crossed my mind, but then the stop_signal would be involved in supervisor::notify which is out of scope for it.  Ultimately, we may want a mid-level abstraction between main and the stop_signal to encapsulate the checkpoint (and more, as my original work on a service-control unit that will control the services state machine and ependencies)